src/pages/[platform]/ai/conversation/tools/index.mdx

```diff
@@ -0,0 +1,266 @@
+
+</Callout>
+
+Here is an example Lambda function handler for our `getWeather` query:
+
+```ts title="amplify/data/getWeather.ts"
+import type { Schema } from "./resource";
+
+export const handler: Schema["getWeather"]["functionHandler"] = async (
+  event
+) => {
+  // This returns a mock value, but you can connect to any API, database, or other service
+  return {
+    value: 42,
+    unit: 'C'
+  };
+}
+```
+
+Lastly, you will need to update your **`amplify/backend.ts`** file to include the newly defined `getWeather` function.
+
+```ts title="amplify/backend.ts"
+// highlight-start
+import { getWeather } from "./data/resource";
+// highlight-end
+
+const backend = defineBackend({
+  auth,
+  data,
+  // highlight-start
+  getWeather
+  // highlight-end
+});
+```
+
+
+## Connect to any AWS Service
+
+You can connect to any AWS service by defining a custom query and calling that service in the function handler. Then you will need to provide the Lambda with the proper permissions to call the AWS service.
+
+```ts title="amplify/backend.ts"
+import { defineBackend } from "@aws-amplify/backend";
+import { auth } from "./auth/resource";
+import { data } from "./data/resource";
+import { storage } from "./storage/resource";
+import { getWeather } from "./functions/getWeather/resource";
+import { PolicyStatement } from "aws-cdk-lib/aws-iam";
+
+const backend = defineBackend({
+  auth,
+  data,
+  storage,
+  getWeather
+});
+
+backend.getWeather.resources.lambda.addToRolePolicy(
+  new PolicyStatement({
+    resources: ["[resource arn]",],
+    actions: ["[action]"],
+  }),
+)
+```
+
+
+
+## Custom Lambda Tools
+
+Conversation routes can also have completely custom tools defined in a Lambda handler.
+
+### Create a custom conversation handler function
+
+```ts title="amplify/data/resource.ts"
+import { type ClientSchema, a, defineData } from '@aws-amplify/backend';
+import { defineConversationHandlerFunction } from '@aws-amplify/backend-ai/conversation';
+
+const chatHandler = defineConversationHandlerFunction({
+  entry: './chatHandler.ts',
+  name: 'customChatHandler',
+  models: [
+    { modelId: a.ai.model("Claude 3 Haiku") }
+  ]
+});
+
+const schema = a.schema({
+  chat: a.conversation({
+    aiModel: a.ai.model('Claude 3 Haiku'),
+    systemPrompt: "You are a helpful assistant",
+    handler: chatHandler,
+  })
+})
+```
+
+### Implement the custom handler
+
+```ts title="amplify/data/chatHandler.ts"
+
+import {
+    ConversationTurnEvent,
+    ExecutableTool,
+    handleConversationTurnEvent,
+} from '@aws-amplify/ai-constructs/conversation/runtime';
+import { ToolResultContentBlock } from '@aws-sdk/client-bedrock-runtime';
+
+const thermometer: ExecutableTool = {
+  name: 'thermometer',
+  description: 'Returns current temperature in a city',
+  execute: (input): Promise<ToolResultContentBlock> => {
+    if (input && typeof input === 'object' && 'city' in input) {
```

Comment:
nit: this line is no longer necessary since https://github.com/aws-amplify/amplify-backend/pull/2070